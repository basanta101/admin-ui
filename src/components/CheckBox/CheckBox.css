/* .checkbox:disabled { 
    cursor: not-allowed;
} */