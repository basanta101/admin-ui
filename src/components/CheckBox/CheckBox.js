import { useState } from 'react';


const CheckBox = ({ checked = false, handleCheckCB = (f) => f }) => {
    const [ checkValue, updateCheckValue] = useState(checked);

    const handleCheck = (ev) => {
       console.log('ppp', ev.target.checked);
       const { checked } = ev.target;
       updateCheckValue(checked);
       handleCheckCB?.(checked);
    }

   return (
       <>
         <input type="checkbox" onChange={handleCheck} defaultChecked={checkValue}/>
       </>
   )
};

export default CheckBox;