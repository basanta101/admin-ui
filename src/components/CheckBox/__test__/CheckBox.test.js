import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import CheckBox from "../CheckBox";


test("user can check uncheck the checkBox", () => {
  let checked = false;
  const handleCheckCb = jest.fn((val) => {
    checked = val;
  });
  const { container, rerender } = render(
    <CheckBox handleCheckCb={handleCheckCb} checked={checked} />
  );
  const input = container.firstChild;
  fireEvent.click(input);
  expect(handleCheckCb).toHaveBeenCalledTimes(1);
  rerender(<CheckBox handleCheckCb={handleCheckCb} checked={checked} />);
  expect(input).toBeChecked();
});

it("renders a snapshot of CheckBox", () => {
  const tree = renderer.create(<CheckBox />).toJSON();
  expect(tree).toMatchSnapshot();
});