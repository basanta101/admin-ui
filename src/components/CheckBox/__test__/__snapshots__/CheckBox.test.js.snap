// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of CheckBox 1`] = `
<input
  checked={false}
  className="checkBox"
  disabled={false}
  onChange={[Function]}
  type="checkbox"
/>
`;
